core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -199,6 +210,25 @@ protected void doStop() throws Exception {
     ServletContextHandler context = new ServletContextHandler(ServletContextHandler.SESSIONS);
     context.setContextPath("/");
     context.addServlet(servletHolder, "/*");
+    String authMethod = getConfiguration().getString(RestConfig.AUTHENTICATION_METHOD);
+    if (!"NONE".equals(authMethod)) {
```

Comment:
i think we can get away with just supporting jaas. you can use [jaas](http://www.eclipse.org/jetty/documentation/9.3.x/jaas-support.html#_a_closer_look_at_jaasloginservice) to config anything else with it (maybe not spnego).

i'll add a check for basic. the other possible value here would be `FORM`, but figuring out configs for that will be a pain equivalent to the constraint mappings http://www.eclipse.org/jetty/documentation/9.4.x/configuring-security-authentication.html#_configuring_an_authentication_mechanism